04.ls/ls.rb

```diff
@@ -8,10 +8,12 @@ def main
   option = OptionParser.new
   options = {}
   option.on('-a') { options[:all] = true }
+  option.on('-r') { options[:reverse] = true }
 
   option.parse!(ARGV)
   filenames = options[:all] ? Dir.glob('*', File::FNM_DOTMATCH) : Dir.glob('*')
-  show_file_list(filenames)
+  filenames_reverse = filenames.reverse if options[:reverse]
+  show_file_list(filenames_reverse || filenames)
```

Comment:
```suggestion
  sorted_filenames = options[:reverse] ? filenames.reverse : filenames
  show_file_list(sorted_filenames)
```

逆順でも普通の並び順でも通用するような抽象的な変数名に入れた方がベターです。
参考 https://bootcamp.fjord.jp/articles/78

逆順と普通の並び順で別々の変数を用意したりすると、どんどん変数が増えていってしまうので。